addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/discovery/KodiUpnpDiscoveryParticipant.java

```diff
@@ -44,11 +49,19 @@
 
     @Activate
     protected void activate(ComponentContext componentContext) {
-        if (componentContext.getProperties() != null) {
-            String autoDiscoveryPropertyValue = (String) componentContext.getProperties().get("enableAutoDiscovery");
-            if (StringUtils.isNotEmpty(autoDiscoveryPropertyValue)) {
-                isAutoDiscoveryEnabled = Boolean.valueOf(autoDiscoveryPropertyValue);
-            }
+        activateOrModifyService(componentContext);
+    }
+
+    @Modified
+    protected void modified(ComponentContext componentContext) {
+        activateOrModifyService(componentContext);
+    }
+
+    private void activateOrModifyService(ComponentContext componentContext) {
+        Dictionary<String, Object> properties = componentContext.getProperties();
```

Comment:
Afaik, you should annotate the Object type `@Nullable`

Reply:
Maybe @maggu2810 knows?

Reply:
Hm, to be honest, I do not know. Just a stupid guess after some google research. But maybe I have not searched deep enough.

Reply:
https://www.osgi.org/release-4-version-4-2-download/
OSGi Service Platform Release 4 Version 4.2 Compendium Specification
e.g. 112.5.8 and 112.5.9
`Component Context` is IMHO always present for a `Component`, similar to `BundleContext` for a `Bundle`.

Reply:
Nice, thanks for the link. It will be a good lecture for my next vacation.

Reply:
ESH needs to be compatible to 4.2 but for a lecture I would prefer the most recent specification. IMHO it is more detailed at some places (and IIRC you can use > 4.2 features in OH).